lib/testing/test_runner.sh

```diff
@@ -357,6 +357,15 @@ teardown()
 
     rm -rf "$STATEDIR"
 
+    for i in /tmp/"${NS}"_PID_*; do
```

Comment:
This won't work, as it will kill all processes, even those that are running in parallel. To support this some global per test run ID needs to be added.

Reply:
I think the right thing here is to change the tmpfile to a directory (with mktemp -d) and dump everything in there; that also makes it easier to find all the pid files for a run...

Reply:
I'm not sure what do you mean by processes running in parallel. If you mean processes from tests that are running in parallel, then they all would have to have a different $NS, wouldn't they?

Reply:
I was under the impression we could run the tests in parallel (or maybe I was hoping for this ;), but looking at the code this is not the case, so we might not need to worry about it for now.